idea/src/org/jetbrains/kotlin/idea/inspections/ConstantConditionIfInspection.kt

```diff
@@ -26,56 +26,76 @@ import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
 class ConstantConditionIfInspection : AbstractKotlinInspection() {
 
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
-        return ifExpressionVisitor(fun(expression) {
-            val condition = expression.condition ?: return
+        return ifExpressionVisitor { expression -> registerProblem(expression, holder) }
+    }
 
-            val context = condition.analyze(BodyResolveMode.PARTIAL_WITH_CFA)
-            val constantValue = condition.constantBooleanValue(context) ?: return
+    private fun registerProblem(expression: KtIfExpression, holder: ProblemsHolder?): List<ConstantConditionIfFix> {
+        val condition = expression.condition ?: return emptyList()
 
-            val fixes = mutableListOf<LocalQuickFix>()
+        val context = condition.analyze(BodyResolveMode.PARTIAL_WITH_CFA)
+        val constantValue = condition.constantBooleanValue(context) ?: return emptyList()
 
-            if (expression.branch(constantValue) != null) {
-                val keepBraces = expression.isElseIf() && expression.branch(constantValue) is KtBlockExpression
-                fixes += SimplifyFix(constantValue, expression.isUsedAsExpression(context), keepBraces)
-            }
+        val fixes = mutableListOf<ConstantConditionIfFix>()
 
-            if (!constantValue && expression.`else` == null) {
-                fixes += RemoveFix()
-            }
+        if (expression.branch(constantValue) != null) {
+            val keepBraces = expression.isElseIf() && expression.branch(constantValue) is KtBlockExpression
+            fixes += SimplifyFix(constantValue, expression.isUsedAsExpression(context), keepBraces)
+        }
+
+        if (!constantValue && expression.`else` == null) {
+            fixes += RemoveFix()
+        }
+
+        holder?.registerProblem(
+            condition,
+            "Condition is always '$constantValue'",
+            *fixes.toTypedArray()
+        )
+
+        return fixes
+    }
+
+    fun applyFixIfNeed(ifExpression: KtIfExpression) {
```

Comment:
I'd extract this and previous function to companion object, if possible. Also I don't like this name too. Something like `applySingleFixIfAny` or `applyFixIfSingle` would be better.